strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/storage/snapshot/MavenSnapshotManager.java

```diff
@@ -0,0 +1,268 @@
+        File file = new File(basePath);
+
+        LayoutProvider layoutProvider = getLayoutProvider(repository, layoutProviderRegistry);
+
+        Metadata metadata = mavenMetadataManager.readMetadata(Paths.get(basePath));
+
+        if (metadata != null && metadata.getVersioning() != null)
+        {
+            /**
+             * map of snapshots for removing
+             * k - number of the build, v - version of the snapshot
+             */
+            Map<Integer, String> mapToRemove = getRemovingSnapshots(metadata, numberToKeep, keepPeriod);
+
+            if (!mapToRemove.isEmpty())
+            {
+                List<String> removingSnapshots = new ArrayList<>();
+
+                new ArrayList<>(mapToRemove.values()).forEach(e -> removingSnapshots.add(metadata.getArtifactId()
+                                                                                                 .concat("-")
+                                                                                                 .concat(e)
+                                                                                                 .concat(".jar"))
+                );
+
+                List<File> list = Arrays.asList(file.listFiles());
+
+                list.stream()
+                    .filter(File::isFile)
+                    .filter(e -> ArtifactUtils.isArtifact(e.getPath()) && removingSnapshots.contains(e.getName()))
+                    .forEach(e ->
+                             {
+                                 try
+                                 {
+
+                                     layoutProvider.delete(storageId, repository.getId(), e.getPath(), true);
+
+                                     String artifactName = e.getName();
+                                     artifactName = artifactName.substring(0, artifactName.lastIndexOf('.'));
+                                     String pomPath = Paths.get(e.getParent(), artifactName)
+                                                           .toString()
+                                                           .concat(".pom");
+
+                                     layoutProvider.delete(storageId, repository.getId(), pomPath, true);
+                                 }
+                                 catch (IOException e1)
+                                 {
+                                     logger.error(e1.getMessage(), e1);
+                                 }
+
+                             });
+                return true;
+            }
+        }
+        return false;
+
+    }
+
+    /**
+     * To get map of removing timestamped snapshots
+     *
+     * @param metadata     type Metadata
+     * @param numberToKeep type int
+     * @param keepPeriod   type int
+     * @return type Map<Integer, String>
+     * @throws IOException
+     * @throws XmlPullParserException
+     */
+    private Map<Integer, String> getRemovingSnapshots(Metadata metadata,
```

Comment:
I think this method should be called `getRemovableTimestampedSnapshots`, as it's more intuitive.